pkg/proctree/proctree_procfs.go

```diff
@@ -0,0 +1,199 @@
+package proctree
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/aquasecurity/tracee/pkg/errfmt"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/pkg/utils"
+	"github.com/aquasecurity/tracee/pkg/utils/proc"
+)
+
+// FeedFromProcFS feeds the process tree with data from procfs.
+func (pt *ProcessTree) FeedFromProcFS() error {
+	procDir := "/proc"
+
+	// getProcessByPID returns a process by its PID.
+	getProcessByPID := func(pid int) (*Process, error) {
+		if pid <= 0 {
+			return nil, errfmt.Errorf("invalid PID")
+		}
+		status, err := proc.NewProcStatus(pid)
+		if err != nil {
+			return nil, errfmt.Errorf("%v", err)
+		}
+		stat, err := proc.NewProcStat(pid)
+		if err != nil {
+			return nil, errfmt.Errorf("%v", err)
+		}
+
+		startTimeNs := utils.ClockTicksToNsSinceBootTime(stat.StartTime)
+		hash := utils.HashTaskID(uint32(status.GetPid()), startTimeNs) // status pid == tid
+
+		return pt.GetOrCreateProcessByHash(hash), nil
+	}
+
+	// dealWithProc deals with a process from procfs.
+	dealWithProc := func(pid int) error {
+		if pid <= 0 {
+			return errfmt.Errorf("invalid PID")
+		}
+		status, err := proc.NewProcStatus(pid)
+		if err != nil {
+			return errfmt.Errorf("%v", err)
+		}
+		stat, err := proc.NewProcStat(pid)
+		if err != nil {
+			return errfmt.Errorf("%v", err)
+		}
+		if status.GetPid() != status.GetTgid() {
+			return errfmt.Errorf("invalid process") // sanity check (process, not thread)
+		}
+		if stat.StartTime == 0 {
+			return errfmt.Errorf("invalid start time")
+		}
+
+		// given process (pid) hash
+		startTimeNs := utils.ClockTicksToNsSinceBootTime(stat.StartTime)
+		hash := utils.HashTaskID(uint32(status.GetPid()), startTimeNs)
+
+		// update tree for the given process
+		process := pt.GetOrCreateProcessByHash(hash)
+		procInfo := process.GetInfo()
+		procInfo.SetFeed(
+			TaskInfoFeed{
+				Tid:         int(status.GetPid()),    // status: pid == tid
+				Pid:         int(status.GetTgid()),   // status: tgid == pid
+				PPid:        int(status.GetPPid()),   // status: ppid == ppid
+				NsTid:       int(status.GetNsTgid()), // status: nstgid == nspid
```

Comment:
Is that correct? From one side I see namsepaced tid, and from the other namespaced tgid